### PR TITLE
Add support for URL search params in WorkPlan components

### PR DESCRIPTION
**Pull request by rahul6603**
Issue - [WorkPlan component should support search Params via the URL #614](https://github.com/I-TECH-UW/OpenELIS-Global-2/issues/614)

### CODE REVIEW

**Comment by rahul6603** [comment]
https://github.com/I-TECH-UW/OpenELIS-Global-2/assets/128216878/9142a7f0-5882-49ed-b9d1-06b83610fbfb

File: frontend/src/components/workplan/TestSectionSelectForm.js [hunk 1]
```diff
@@ -20,7 +22,18 @@ function TestSectionSelectForm(props) {
 
   useEffect(() => {
     mounted.current = true;
-    getFromOpenElisServer("/rest/user-test-sections", getTestUnits);
+    let testSectionId = new URLSearchParams(window.location.search).get(
```
Comment (mozzy11) [inline frontend/src/components/workplan/TestSectionSelectForm.js:25]: rename searchParam to  `testSectionId` to be consisted with the old UI

File: frontend/src/components/workplan/TestSelectForm.js [hunk 2]
```diff
@@ -20,7 +22,18 @@ function TestSelectForm(props) {
 
   useEffect(() => {
     mounted.current = true;
-    getFromOpenElisServer("/rest/test-list", getTests);
+    let testId = new URLSearchParams(window.location.search).get(
+      "test_id"
```
Comment (mozzy11) [inline frontend/src/components/workplan/TestSelectForm.js:26]: rename to `testId`  to be consisted with old UI

File: frontend/src/components/workplan/PanelSelectForm.js [hunk 3]
```diff
@@ -20,7 +22,18 @@ function PanelSelectForm(props) {
 
   useEffect(() => {
     mounted.current = true;
-    getFromOpenElisServer("/rest/panels", getPanels);
+    let panelId = new URLSearchParams(window.location.search).get(
+      "panel_id"
```
Comment (mozzy11) [inline frontend/src/components/workplan/PanelSelectForm.js:26]: rename to `paneId`

**Comment by mozzy11** [comment]
Thanks @rahul6603 , generally looks good.
addded a few comments above .
we generally use Camel Cased params as opposed to Snake_Cased params

File: frontend/src/components/workplan/TestSectionSelectForm.js (hunk 1 above)
Comment (rahul6603) [reply frontend/src/components/workplan/TestSectionSelectForm.js:25]: Thanks, updated to testSectionId

File: frontend/src/components/workplan/TestSelectForm.js (hunk 2 above)
Comment (rahul6603) [reply frontend/src/components/workplan/TestSelectForm.js]: Thanks, updated to testId

File: frontend/src/components/workplan/PanelSelectForm.js (hunk 3 above)
Comment (rahul6603) [reply frontend/src/components/workplan/PanelSelectForm.js]: Thanks, updated to paneId

**Comment by rahul6603** [comment]
> Thanks @rahul6603 , generally looks good. addded a few comments above . we generally use Camel Cased params as opposed to Snake_Cased params

@mozzy11 makes sense, have updated to use camel cased params

**Review by mozzy11 — APPROVED** [review]

**Pull request merged**